// jobsuche-api-js/lib/cjs/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.fetchCompanyInfo = exports.getCompanyLogoURL = exports.authManager = exports.fetchJobDetailsV3 = exports.fetchJobDetailsV2 = exports.searchParamsExample = exports.jobsSearchOld = exports.jobsSearch = exports.getJwt = void 0;
const authApi_1 = __importDefault(require("./auth/authApi"));
exports.getJwt = authApi_1.default;
const searchParamsExample_1 = require("./examples/searchParamsExample");
Object.defineProperty(exports, "searchParamsExample", { enumerable: true, get: function () { return searchParamsExample_1.searchParamsExample; } });
const jobsSearch_1 = require("./api/jobsSearch");
Object.defineProperty(exports, "jobsSearchOld", { enumerable: true, get: function () { return jobsSearch_1.jobsSearchOld; } });
Object.defineProperty(exports, "jobsSearch", { enumerable: true, get: function () { return jobsSearch_1.jobsSearch; } });
const jobDetails_1 = require("./api/jobDetails");
Object.defineProperty(exports, "fetchJobDetailsV3", { enumerable: true, get: function () { return jobDetails_1.fetchJobDetailsV3; } });
Object.defineProperty(exports, "fetchJobDetailsV2", { enumerable: true, get: function () { return jobDetails_1.fetchJobDetailsV2; } });
const authManager_1 = __importDefault(require("./auth/authManager"));
exports.authManager = authManager_1.default;
const companyLogo_1 = require("./api/companyLogo");
Object.defineProperty(exports, "getCompanyLogoURL", { enumerable: true, get: function () { return companyLogo_1.getCompanyLogoURL; } });
const companyInfo_1 = require("./api/companyInfo");
Object.defineProperty(exports, "fetchCompanyInfo", { enumerable: true, get: function () { return companyInfo_1.fetchCompanyInfo; } });
