// jobsuche-api-js/lib/cjs/auth/authApi.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const axios_1 = __importDefault(require("axios"));
const querystring_1 = __importDefault(require("querystring"));
const urls_1 = require("../constants/urls");
async function getJwt() {
    const data = querystring_1.default.stringify({
        client_id: 'c003a37f-024f-462a-b36d-b001be4cd24a',
        client_secret: '32a39620-32b3-4307-9aa1-511e3d7f48a8',
        grant_type: 'client_credentials',
    });
    try {
        const response = await axios_1.default.post(`https://${urls_1.host}${urls_1.authSuffix}`, data);
        return response.data.access_token;
    }
    catch (error) {
        if (axios_1.default.isAxiosError(error)) {
            console.error('Axios error:', error.message);
            console.error('Error details:', error.toJSON());
        }
        else {
            console.error('Unexpected error:', error);
        }
        return {
            error: error,
            clientId: process.env.CLIENT_ID,
            clientSecret: process.env.CLIENT_SECRET,
        };
        // throw error;
    }
}
exports.default = getJwt;
