// jobsuche-api-js/lib/cjs/constants/urls.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.companyInfoLink = exports.companyLogoV3 = exports.https = exports.jobDetailV3Link = exports.jobDetailV2Link = exports.logoLink = exports.jobLink = exports.authSuffix = exports.host = exports.headers = void 0;
const host = "rest.arbeitsagentur.de";
exports.host = host;
const headers = {
    "X-Api-Key": "jobboerse-jobsuche",
};
exports.headers = headers;
const https = "https://";
exports.https = https;
const serverSuffix = "jobboerse/jobsuche-service";
const serviceSuffix = "vermittlung/ag-darstellung-service";
const jobSearchPrefix = https + host + "/" + serverSuffix;
const jobsBase = "/pc/v4/jobs";
const companyInfoBase = "/pc/v1/arbeitgeberdarstellung/";
const serviceInfoPrefix = https + host + "/" + serviceSuffix;
/**
 * @info This one doesn't require authentication with headers.
 */
const logoBaseED = "/ed/v1/arbeitgeberlogo/";
const jobDetailsV2 = "/pc/v2/jobdetails/";
const jobDetailsV3 = "/pc/v3/jobdetails/";
/**
 * @info This one doesn't require authentication with headers.
 */
const logoBaseCT = "/ct/v1/arbeitgeberlogo";
// Company Logo:
const authSuffix = "/oauth/gettoken_cc";
exports.authSuffix = authSuffix;
// Full links
const jobLink = jobSearchPrefix + jobsBase;
exports.jobLink = jobLink;
const logoLink = jobSearchPrefix + logoBaseED;
exports.logoLink = logoLink;
const jobDetailV2Link = jobSearchPrefix + jobDetailsV2;
exports.jobDetailV2Link = jobDetailV2Link;
const jobDetailV3Link = jobSearchPrefix + jobDetailsV3;
exports.jobDetailV3Link = jobDetailV3Link;
const companyLogoV3 = serviceInfoPrefix + logoBaseCT;
exports.companyLogoV3 = companyLogoV3;
const companyInfoLink = serviceInfoPrefix + companyInfoBase;
exports.companyInfoLink = companyInfoLink;
